src/lib/Nbe.ml

```diff
@@ -1023,8 +1023,13 @@ struct
     | D.Dim0 -> ret S.Dim0
     | D.Dim1 -> ret S.Dim1
     | D.DimVar lvl ->
-      let+ ix = quote_var lvl in
-      S.Var ix
+      let* eq0 = lift_cmp @@ CmpM.test_sequent [Cof.Cof (Cof.Eq(D.Dim0, D.DimVar(lvl)))] Cof.top in 
```

Comment:
Good guess! I think this will work if you flip something. The meaning of `test_sequent [phi ...] psi` is to see whether the conjunction of `phi ...` implies `psi`. But everything implies `top`.

I think you will want to do `test_sequent [] (i=0)` etc.